### PR TITLE
[#165356866] review-mixpanel-spid-login-trace

### PR DESCRIPTION
**Pull request by Undermaken**
review some debts from PR #952

### CODE REVIEW

**Comment by digitalcitizenship** [comment]
## Affected stories

  * ⚙️ [#165356866](https://www.pivotaltracker.com/story/show/165356866): Mixpanel Login trace - recuperare tutte le review segnalate e non incluse nella PR #952 

<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: ts/screens/authentication/IdpLoginScreen.tsx [hunk 1]
```diff
@@ -124,22 +123,16 @@ class IdpLoginScreen extends React.Component<Props, State> {
...
     });
 
-    onNavigationStateChange(
-      () => {
-        this.props.dispatchLoginFailed(loginFailure());
-      },
-      token => {
-        this.props.dispatchLoginSuccess(loginSuccess(token));
-      }
+    onNavigationStateChange(this.props.dispatchLoginFailed, token =>
```
Comment (cloudify) [inline ts/screens/authentication/IdpLoginScreen.tsx]:
```suggestion
    onNavigationStateChange(this.props.dispatchLoginFailed, this.props.dispatchLoginSuccess)
```

File: ts/screens/authentication/IdpLoginScreen.tsx [hunk 2]
```diff
@@ -229,9 +222,9 @@ const mapStateToProps = (state: GlobalState) => ({
 });
 
 const mapDispatchToProps = (dispatch: Dispatch) => ({
-  dispatchUrlChanged: (action: Action) => dispatch(action),
-  dispatchLoginSuccess: (action: Action) => dispatch(action),
-  dispatchLoginFailed: (action: Action) => dispatch(action)
+  dispatchUrlChanged: (url: string) => dispatch(idpLoginUrlChanged({ url })),
+  dispatchLoginSuccess: (token: SessionToken) => dispatch(loginSuccess(token)),
+  dispatchLoginFailed: () => dispatch(loginFailure())
```
Comment (cloudify) [inline ts/screens/authentication/IdpLoginScreen.tsx]:
```suggestion
  dispatchLoginFailure: () => dispatch(loginFailure())
```

File: ts/screens/authentication/IdpLoginScreen.tsx [hunk 3]
```diff
@@ -229,9 +222,9 @@ const mapStateToProps = (state: GlobalState) => ({
 });
 
 const mapDispatchToProps = (dispatch: Dispatch) => ({
-  dispatchUrlChanged: (action: Action) => dispatch(action),
-  dispatchLoginSuccess: (action: Action) => dispatch(action),
-  dispatchLoginFailed: (action: Action) => dispatch(action)
+  dispatchUrlChanged: (url: string) => dispatch(idpLoginUrlChanged({ url })),
```
Comment (cloudify) [inline ts/screens/authentication/IdpLoginScreen.tsx]:
```suggestion
  dispatchIdpLoginUrlChanged: (url: string) => dispatch(idpLoginUrlChanged({ url })),
```

**Pull request merged**